File: examples/v2/task/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
task_suffix = config.get("taskSuffix") or "dev"
task_id = config.get("taskID") or "example-task-id"

pool = ps_v2.agents.Pool(
    "pool",
    org_name=service_org,
    name=f"v2-task-pool-{task_suffix}",
    description="Pool used by the v2 task example",
)

ps_v2.agents.Task(
    "task",
    org_name=service_org,
    task_id=task_id,
    approval_mode="manual",
    permission_mode="default",
    source="api",
    plan_mode=False,
)

pulumi.export("poolName", pool.name)

File: examples/v2/task/go/main.go
package main

import (
	agents "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/agents"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		taskSuffix := cfg.Get("taskSuffix")
		if taskSuffix == "" {
			taskSuffix = "dev"
		}
		taskID := cfg.Get("taskID")
		if taskID == "" {
			taskID = "example-task-id"
		}

		pool, err := agents.NewPool(ctx, "pool", &agents.PoolArgs{
			OrgName:     pulumi.String(serviceOrg),
			Name:        pulumi.String("v2-task-pool-" + taskSuffix),
			Description: pulumi.String("Pool used by the v2 task example"),
		})
		if err != nil {
			return err
		}

		if _, err := agents.NewTask(ctx, "task", &agents.TaskArgs{
			OrgName:        pulumi.String(serviceOrg),
			TaskID:         pulumi.String(taskID),
			ApprovalMode:   pulumi.String("manual"),
			PermissionMode: pulumi.String("default"),
			Source:         pulumi.String("api"),
			PlanMode:       pulumi.Bool(false),
		}); err != nil {
			return err
		}

		ctx.Export("poolName", pool.Name)
		return nil
	})
}
